lightbug_http/header.mojo

```diff
@@ -837,31 +851,31 @@ struct headerScanner:
             raise Error("Invalid header, did not find a colon")
         
         var jump_to = colon + 1
-        self.set_key(self.b()[:jump_to])
+        self.set_key(self.__b[:colon])
 
-        while len(self.b()) > jump_to and (self.b()[jump_to] == bytes(whitespace, pop=False)[0]):
+        while len(self.__b) > jump_to and (self.__b[jump_to] == whitespace_byte):
             jump_to += 1
             self.set_next_line(self.next_line() - 1)
         
         self.set_subslice_len(self.subslice_len() + jump_to)
-        self.set_b(self.b()[jump_to:])
+        self.set_b(self.__b[jump_to:])
 
         if self.next_line() >= 0:
             jump_to = self.next_line()
             self.set_next_line(-1)
         else:
-            jump_to = index_byte(self.b(), bytes(nChar, pop=False)[0])
+            jump_to = index_byte(self.__b, nChar_byte)
         if jump_to < 0:
             raise Error("Invalid header, did not find a newline")
         
         jump_to += 1
-        self.set_value(self.b()[:jump_to])
+        self.set_value(self.__b[:jump_to-2]) # -2 to exclude the \r\n
```

Comment:
is this needed now after you found the issue with the date header? or is this not related

Reply:
I think it's still needed since the scanner seems to include the `\r\n` at the end of the each line of headers.